test_util/cluster.py

```diff
@@ -0,0 +1,260 @@
+import itertools
+import logging
+import os
+import random
+import stat
+import string
+from collections import namedtuple
+from contextlib import closing, contextmanager
+from os.path import join
+
+import passlib.hash
+import pkg_resources
+from retrying import retry
+
+import test_util.ccm
+import test_util.installer_api_test
+import test_util.test_runner
+from ssh.ssh_tunnel import SSHTunnel, TunnelCollection
+
+
+Host = namedtuple('Host', ['public_ip', 'private_ip'])
+
+
+class TestCluster:
+
+    def __init__(
+            self,
+            public_ips,
+            ssh_user,
+            ssh_key_path,
+            remote_dir,
+            num_masters,
+            num_agents,
+            num_public_agents,
+            setup=True,
+    ):
+        self.vpc = None
```

Comment:
why not have people pass in teh vpc. The VPC object should implement a basic interface:
1. get_hosts()
2. get_ssh_info()
3. delete()